src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -646,6 +807,35 @@ def get_synopsys_rm_tarball(self, product: str, settings_key: str = "") -> str:
         else:
             return synopsys_rm_tarball
 
+class MentorTool(HammerTool):
+    """ Mix-in trait with functions useful for Mentor-Graphics-based tools. """
+    @property
+    def env_vars(self) -> Dict[str, str]:
+        """
+        Get the list of environment variables required for this tool.
+        Note to subclasses: remember to include variables from super().env_vars!
+        """
+        return {
+            "MGLS_LICENSE_FILE": self.get_setting("mentor.MGLS_LICENSE_FILE")
```

Comment:
Note to subclasses: remember to include variables from super().env_vars!

Reply:
Good catch.

Reply:
What it means is that probably we need to do something like

```python
@property
def env_var(self) -> Dict[str, str]:
  result = dict(super().env_vars)
  result.update({"MGLS_LICENSE_FILE": "..."})
  return result
```

Reply:
OK. We'll probably want to fix SynopsysTool as well: https://github.com/ucb-bar/hammer/blob/60f404d155c347cf8efae548d81a933189d0c470/src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py#L771-L780

Reply:
Opened https://github.com/ucb-bar/hammer/pull/214